### PR TITLE
Allows genemodders to become most off-ship roles.

### PR DESCRIPTION
**Pull request by EmberShinori**
Adds genemodders to unrestricted_species so that they're allowed to be most off-ship roles. Does not allow them to join the Salvage Gantry since that's not wanted, I believe. Also blacklists genemodders from selecting Ascent since genemodder cosmetics aren't removed upon spawning in as Ascent.

### CODE REVIEW

File: code/__defines/mobs.dm [hunk 1]
```diff
@@ -307,7 +307,7 @@
 #define SPECIES_BOOSTER     "Booster"
 #define SPECIES_MONKEY      "Monkey"
 
-#define UNRESTRICTED_SPECIES list(SPECIES_HUMAN, SPECIES_DIONA, SPECIES_IPC, SPECIES_UNATHI, SPECIES_SKRELL, SPECIES_TRITONIAN, SPECIES_SPACER, SPECIES_VATGROWN, SPECIES_GRAVWORLDER, SPECIES_BOOSTER, SPECIES_MULE)
+#define UNRESTRICTED_SPECIES list(SPECIES_HUMAN, SPECIES_DIONA, SPECIES_IPC, SPECIES_UNATHI, SPECIES_SKRELL, SPECIES_TRITONIAN, SPECIES_SPACER, SPECIES_VATGROWN, SPECIES_GRAVWORLDER, SPECIES_BOOSTER, SPECIES_MULE, SPECIES_CUSTOM)
```
Comment (BurpleBineapple) [inline code/__defines/mobs.dm]:
```suggestion
#define UNRESTRICTED_SPECIES list(SPECIES_HUMAN, SPECIES_DIONA, SPECIES_IPC, SPECIES_UNATHI, SPECIES_SKRELL, SPECIES_TRITONIAN, SPECIES_SPACER, SPECIES_VATGROWN, SPECIES_GRAVWORLDER, SPECIES_BOOSTER, SPECIES_MULE, SPECIES_CUSTOM, SPECIES_SHELL)
```

**Review by BurpleBineapple — CHANGES_REQUESTED** [review]
If you could add Shell IPCs to this list, that would be greatly appreciated as well

File: code/__defines/mobs.dm (hunk 1 above)
Comment (Vizoplez) [inline code/__defines/mobs.dm]: Also SPECIES_ADHERENT too please.

**Review by io-util — APPROVED** [review]
Looks alright from my side, @BurpleBineapple.

**Review by BurpleBineapple — APPROVED** [review]

**Pull request merged**